### ISSUE TITLE
Broken inference on conditional str.__iadd__?

### ISSUE THREAD

**Issue by xmo-odoo** [PyCQA/pylint#813]
```python
from astroid.builder import parse

m = parse('''
a = ''
if cond:
    a += thing()
print a
''')
```
`m.body[-1].values[0].infer()` will only contain a `Const(value='')`, completely ignoring the augmented assignment. I would have expected something similar to a regular concatenation-and-assignment:
```python
from astroid.builder import parse

m = parse('''
a = ''
if cond:
    a = a + thing()
print a
''')
print list(m.body[-1].values[0].infer())
```
```
[<Const(str) l.2 [] at 0x1020d7550>, YES]
```
since augmented assignment will fallback on a normal operate-then-assign if the types are not augmented-assignment compatible:
```python
>>> class Foo(object):
...     def __radd__(self, other):
...             return self
... 
>>> a = 'a'
>>> a += Foo()
>>> a
<__main__.Foo object at 0x1020c0a10>
```

**Comment by xmo-odoo** [comment]
Wrong repository, sorry, this should probably be on the astroid repo
